File: backend/models/HRInterviewReport.js
const mongoose = require('mongoose');

const hrInterviewReportSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    index: true
  },
  interviewId: {
    type: String,
    required: true,
    index: true
  },
  interviewType: {
    type: String,
    default: 'HR'
  },
  questions: {
    type: Array,
    default: []
  },
  answers: {
    type: Array,
    default: []
  },
  aiFeedback: {
    type: Array,
    default: []
  },
  strengths: {
    type: Array,
    default: []
  },
  weaknesses: {
    type: Array,
    default: []
  },
  // HR-specific scores
  communicationScore: {
    type: Number,
    default: 0
  },
  confidenceScore: {
    type: Number,
    default: 0
  },
  professionalismScore: {
    type: Number,
    default: 0
  },
  overallScore: {
    type: Number,
    default: 0
  },
  recommendation: {
    type: String,
    default: ''
  },
  duration: {
    type: Number,  // seconds
    default: 0
  },
  createdByAI: {
    type: Boolean,
    default: true
  }
}, { timestamps: true });

module.exports = mongoose.model(
  'HRInterviewReport',
  hrInterviewReportSchema,
  'HRInterviewReports'
);

File: backend/models/hr_interview_report.py
"""models/hr_interview_report.py — HRInterviewReport table (replaces Mongoose HRInterviewReport.js)"""
from sqlalchemy import Column, Integer, String, Float, Boolean, DateTime, ForeignKey, Text, JSON
from sqlalchemy.sql import func
from database import Base


class HRInterviewReport(Base):
    __tablename__ = "hr_interview_reports"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    interview_id = Column(String(255), nullable=False, index=True)
    interview_type = Column(String(100), default="HR")
    questions = Column(JSON, default=list)
    answers = Column(JSON, default=list)
    ai_feedback = Column(JSON, default=list)
    strengths = Column(JSON, default=list)
    weaknesses = Column(JSON, default=list)
    communication_score = Column(Float, default=0)
    confidence_score = Column(Float, default=0)
    professionalism_score = Column(Float, default=0)
    overall_score = Column(Float, default=0)
    recommendation = Column(Text, default="")
    duration = Column(Integer, default=0)  # seconds
    created_by_ai = Column(Boolean, default=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
